### PR TITLE
Skip IPv6 tests when TCLSH version is < 8.6

### PR DESCRIPTION
**Pull request by enjoy-binbin**
In #786, we did skip it in the daily, but not for the others.
When running ./runtest on MacOS, we will get the failure.
```
couldn't open socket: host is unreachable (nodename nor servname provided, or not known)
```

TCL 8.5 doesn't support ipv6, skip tests tagged with ipv6.
This also revert #786.

### CODE REVIEW

File: tests/unit/cluster/announce-client-ip.tcl [hunk 1]
```diff
@@ -147,3 +148,4 @@ start_cluster 2 2 {tags {external:skip cluster ipv6} overrides {cluster-replica-
         [lindex $clients $j] close
     }
 }
+} ;# if uname
```
Comment (enjoy-binbin) [inline tests/unit/cluster/announce-client-ip.tcl]:
```suggestion
} ;# if uname

```

**Review by madolson — APPROVED** [review]

**Review by hpatro — APPROVED** [review]
Thanks must have been bothering devs using MacOS as well. Why does it not work on MacOS though?

**Comment by madolson** [comment]
> Thanks must have been bothering devs using MacOS as well. Why does it not work on MacOS though?

By default macos has tcl 8.5 installed, which doesn't support ipv6. If you do `brew install tcl-tk` it does work on 8.6. I guess we could skip it based off of that?

File: tests/unit/cluster/announce-client-ip.tcl [hunk 2]
```diff
@@ -68,6 +68,7 @@ start_cluster 2 2 {tags {external:skip cluster} overrides {cluster-replica-no-fa
     }
 }
 
+if {![string match {*Darwin*} [exec uname -a]]} {
```
Comment (madolson) [inline tests/unit/cluster/announce-client-ip.tcl]:
```suggestion
if {$::tcl_version >= 8.6} {
```
I don't get TCL, but we also need `$::tcl_version = $tcl_version` with all the other global defines. In instances.tcl.

**Review by zuiderkwast — COMMENTED** [review]
Can we add the check wherever tags are checked in the test framework, instead of in this test suite?

If tcl < 8.6, skip tests tagged with ipv6.

File: tests/unit/cluster/announce-client-ip.tcl (hunk 2 above)
Comment (enjoy-binbin) [reply tests/unit/cluster/announce-client-ip.tcl]: ok, i verified it, we are good to go.

**Review by zuiderkwast — APPROVED** [review]

**Pull request merged**